R/rep_sample_n.R

```diff
@@ -119,11 +125,18 @@ rep_slice_sample <- function(.data, n = NULL, prop = NULL, replace = FALSE,
     min_val = 0
   )
   check_type(replace, is_truefalse, "TRUE or FALSE")
+  eval_weight_by <- try(rlang::eval_tidy(weight_by), silent = TRUE)
```

Comment:
This is scrappier than I'd like it to be. If there's an unquoted argument passed here that _can_ evaluate to something but is not in the data (try `rlang::eval_tidy(weights)`), then `check_cols` is skipped. Thankfully, in that case, the other check errors well (and in the way it used to):

``` r
library(infer)
library(tibble)

df <- tibble(
   id = 1:5,
   letter = factor(c("a", "b", "c", "d", "e")),
   wts = c(.5, .4, .3, .2, .1)
)

rep_slice_sample(df, n = 2, reps = 5, weight_by = weights)
#> Error: `weight_by` must be 'a numeric vector with length `nrow(.data)` = 5 or an unquoted column name', not 'closure'.
```

<sup>Created on 2023-03-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Also, the single quotes in these error messages are, I believe, new, but our regexes in tests didn't catch that change. We can fix that with https://github.com/tidymodels/infer/issues/482, and test with snapshots in the future to prevent that happening later on.

Reply:
Yeah, I'm surprised to see those. Looks like #482 will address that.

It's neat for me to see the effect of a package being a Posit-supported. Immensely helpful for the propagation of best-practices.